contracts/sophia.md

```diff
@@ -317,14 +317,16 @@ To register a new oracle answering questions of type `'a` with answers of type `
 call `Oracle.register`:
 ```
 Oracle.register(acct : address
-                sign : signature,   // Signed account address
+                sign : signature,   // Signed account address + contract address
                 qfee : int,
                 ttl  : Chain.ttl) : oracle('a, 'b)
 ```
 
 * The `acct` is the address of the oracle to register (can be the same as the contract).
-* The `sign` is a signature of the address to register proving you have the private key
-  of the oracle, or the integer 0 when address is the same as the contract.
+* The `sign` is a signature proving that the contract is allowed to register the account -
+  the account address + the contract address (concatenated as byte arrays) is signed with the
+  private key of the account, proving you have the private key of the oracle to be. If the
+  address is the same as the contract `sign` is 0.
```

Comment:
0 is not a valid signature. The nicest solution is probably to make the signature a named argument, but for now maybe change to "If the address is the same as the contract the signature is ignored".